### PR TITLE
Add recipe for cpprate

### PR DESCRIPTION
**Pull request by johnlees**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by johnlees** [comment]
@conda-forge/help-c-cpp ready for review

**Comment by johnlees** [comment]
https://github.com/orgs/conda-forge/teams/help-c-cpp another request for review

File: recipes/cpprate/build.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+mkdir build
+pushd build
+cmake -DCMAKE_BUILD_EXECUTABLE=1 ..
```
Comment (xhochy) [inline recipes/cpprate/build.sh:3]:
```suggestion
cmake ${CMAKE_ARGS} -DCMAKE_BUILD_EXECUTABLE=1 ..
```

File: recipes/cpprate/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - cmake >=3.1
+    - make
+  host:
+    - llvm-openmp        # [osx]
+    - libgomp            # [linux64]
+    - openblas
```
Comment (xhochy) [inline recipes/cpprate/meta.yaml:24]: To allow for switching blas backends
```suggestion
    - libblas
```

File: recipes/cpprate/build.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+make -j
```
Comment (xhochy) [inline recipes/cpprate/build.sh:4]:
```suggestion
make -j${CPU_COUNT}
```

File: recipes/cpprate/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - eigen
+    - bzip2
+    - xz
+  run:
+    - zlib
```
Comment (xhochy) [inline recipes/cpprate/meta.yaml:29]: Should not be needed but covered by run_exports 
```suggestion
```

File: recipes/cpprate/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    - bzip2
+    - xz
+    - libopenblas
```
Comment (xhochy) [inline recipes/cpprate/meta.yaml:32]:
```suggestion
```

File: recipes/cpprate/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    - libgfortran-ng        # [linux64]
```
Comment (xhochy) [inline recipes/cpprate/meta.yaml:33]: This indicates that you are missing a fortran compiler.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/cpprate/meta.yaml (hunk 6 above)
Comment (johnlees) [reply recipes/cpprate/meta.yaml]: fortran actually not needed, this was copied from another blas recipe, can remove these libraries

File: recipes/cpprate/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    - make
+  host:
+    - llvm-openmp        # [osx]
+    - libgomp            # [linux64]
+    - libblas
+    - eigen
+    - bzip2
+    - xz
+  run:
+    - libcxx             # [osx]
```
Comment (xhochy) [inline recipes/cpprate/meta.yaml:29]: Not needed anymore, the bug that required this was fixed.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/cpprate/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+about:
+  home: https://github.com/tmaklin/cpprate
+  summary: 'Variable Selection in Black Box Methods with RelATive cEntrality (RATE) Measures'
+  description: |
+    cpprate is a reimplementation of https://github.com/lorinanthony/rate in C++.
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file:
+    - LICENSE
+    - external/bs_thread_pool/LICENSE.txt
```
Comment (carterbox) [inline recipes/cpprate/meta.yaml:45]: Please include the licenses for the external header only libraries that CMake is fetching during the build process! https://github.com/tmaklin/cpprate/tree/main/config

Dynamically link if any of these are not header-only.

**Review by carterbox — CHANGES_REQUESTED** [review]
Missing license files for revended dependencies.

**Comment by johnlees** [comment]
@conda-forge/help-c-cpp ready for review

I have removed the unnecessary OS X library, and also added the LICENSE files for the two static libraries which were being included but their licenses missing before (others e.g. blas, openmp are all conda managed I think)

**Review by xhochy — APPROVED** [review]

**Comment by johnlees** [comment]
@carterbox can this be merged now?

**Pull request merged**